feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c

```diff
@@ -275,6 +275,21 @@ bool target_stats_survey_convert(radio_entry_t *radio_cfg, radio_scan_type_t sca
 {
 	target_survey_record_t delta;
 
+	if (data_new->info.chan != data_old->info.chan) {
+		/* Do not consider the old channel's data */
+		memset(data_old, 0, sizeof(*data_old));
+	} else if ((data_new->duration_ms <= data_old->duration_ms) &&
+			(data_new->chan_busy <= data_old->chan_busy)) {
+		/*
+		 * Take care of survey data from CHAN_INFO events which are Read-On-Clear
+		 * in nature, whereas survey data from BSS_CHAN_INFO events are Read only.
+		 * Chan_info event down in the driver updates only the duration, noise floor
```

Comment:
Why are we getting two different kinds of events?  If the Chan_info event only contains duration, noise floor and chan_busy, then all the other counters will be zero and will get reported as such.  This willl cause an awkward discontinuity in the graphs.

Since we are asking for and expecting the BSS_CHAN_INFO wouldn't be better to report that only?  Although, this might be tough if the other event is occurring and resetting counters.

Reply:
We still don't know why do we get the chan_info events. I do not see any code requesting for this event, and it is not that this event is always generated for the current operating channel. It may be that the firmware is auto-generating these events based on some criteria or a bug.

At the driver level (in wmi.c), we see some valid info in the other parameters of chan_info event. I mean, we do see the rx and rx_self info being populated by the firmware. But, the same is not propagated all the way to the user-space. We do not see much of a variation due to this, as the measurement time for chan_info always seemed to be 150ms. So, we are stretching the missing parameters like rx and rx_self for another 150ms using the same old values we read from bss_chan_info.

I still believe, the right thing to do here is to patch the driver to update all the parameters under chan_info event processing. But for some reason, John does not like that idea. I did not understand the reason for not liking it though.